smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy

```diff
@@ -205,6 +170,11 @@ structure httpApiKeyAuth {
     scheme: NonEmptyString,
 }
 
+@trait(selector: "structure > member :not(> :test(union, structure > :test([trait|required])))",
```

Comment:
This selector is out of sync with [the design doc](https://github.com/awslabs/smithy/pull/920/files#diff-f7be63c09c569449e47b6578b6f337c203a2e3d0a176eb58e1fa083adf8b7387R264), but I believe that's more on the doc side than here.